### PR TITLE
Improvements/bug fix for removeStrategy

### PR DESCRIPTION
**Pull request by tomlinton**
Fixes https://github.com/OriginProtocol/origin-dollar/issues/324 and https://github.com/OriginProtocol/origin-dollar/issues/322.

### CODE REVIEW

**Comment by tomlinton** [comment]
This requires an upgrade of Vault. I didn't add it as not sure if its worth it just for these changes. Hopefully it'll make its way into an upgrade shortly.

File: contracts/test/vault/z_compound.js [hunk 1]
```diff
@@ -365,6 +365,12 @@ describe("Vault with Compound strategy", function () {
     expect(await viewVault.totalValue()).to.approxEqual(
       utils.parseUnits("230", 18)
     );
+
+    // Shold be able to add Strategy back. Proves the struct in the mapping
```
Comment (franckc) [inline contracts/test/vault/z_compound.js:369]: typo: Should

**Review by franckc — APPROVED** [review]
LGTM

Perhaps we can plan on a VaultAdmin and VaultCore deploy early next week to pick various changes including this one that have been made to the vault?

File: contracts/test/vault/z_compound.js [hunk 2]
```diff
@@ -366,7 +366,7 @@ describe("Vault with Compound strategy", function () {
       utils.parseUnits("230", 18)
     );
 
-    // Shold be able to add Strategy back. Proves the struct in the mapping
+    // Should be able to add Strategy back. Proves the struct in the mapping
```
Comment (micahalcorn) [inline contracts/test/vault/z_compound.js:369]: 👏

**Review by micahalcorn — APPROVED** [review]

**Pull request merged**